test/style/filter.test.cpp

```diff
@@ -76,13 +93,27 @@ TEST(Filter, Any) {
     ASSERT_TRUE(filter("[\"any\", [\"==\", \"foo\", 0], [\"==\", \"foo\", 1]]", {{ std::string("foo"), int64_t(1) }}));
 }
 
+TEST(Filter, AnyExpression) {
+    ASSERT_FALSE(filter("[\"any\"]"));
+    ASSERT_TRUE(filter("[\"any\", true]"));
+    ASSERT_TRUE(filter("[\"any\",true, false]"));
+    ASSERT_TRUE(filter("[\"any\", true, true]"));
+}
```

Comment:
I think these will still end up testing the non-expression path -- how about just using the ones from the test immediately above, but replacing `"foo"` with `["get", "foo"]`?